scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala

```diff
@@ -206,6 +208,19 @@ case class ScalafmtConfig(
     continuationIndent.defnSite
   )
 
+  private lazy val regexFileOverride = fileOverride.values.map {
+    case (pattern, conf) =>
+      val style = ScalafmtConfig.configReader(this).read(conf)
+      pattern.r.pattern -> style
```

Comment:
What do you think about using `java.nio.file.PathMatcher` instead of regexes? PathMatcher doesn't require escaping dots (`\.`) and it works uniformly across unix/windows. I've been planning for a while to add a PathMatcher-based alternative to `project.includeFilters`

Reply:
`PathMatcher.matches()` requires the argument to be of type `java.nio.file.Path`, which might introduce complications, but I think it would be worthwhile to try and make it work

Reply:
will try

Reply:
ah, wait, how does one instantiate a `PathMatcher`? :) it looks it's a trait. i doubt `metaconfig` does it automatically, right?

Reply:
Metaconfig doesn't support `PathMatcher` out of the box, although it probably should 

```scala
java.nio.file.FileSystems.getDefault().getPathMatcher("glob:**.txt").matches(java.nio.file.Paths.get("foo.txt"))
```

Reply:
good, thank you, done. uploading a new patch shortly.